sdk/js/src/ConfigRequestConsolidator.ts

```diff
@@ -75,6 +75,9 @@ export class ConfigRequestConsolidator {
             .catch((err) => {
                 resolvers.forEach(({ reject }) => reject(err))
             })
+            .finally(() => {
+                this.currentPromise = null
```

Comment:
any concerns about what changing the order of these operations will do?

Reply:
since the `currentPromise` will not execute if there's already an in-flight request, I don't see any changes with this format... the other option is to explicitly move this statement within the `catch` (line 75) and leave the original code as-is, which would eliminate the `.finally` here

Reply:
well now you're resolving all the waiting resolvers before setting currentPromise back to null, whereas before it was null first

Reply:
hm, i'll switch it to set it to null before the resolvers.. because if there's any errors in the resolvers, then it could potentially delay subsequent requests

Reply:
updated to explicitly reset the `currentPromise` in the catch handler

Reply:
oh nvm diff is weird

Reply:
it would be cleaner to just try catch inside performRequest

Reply:
sounds good, that way we can avoid having the reset behaviour split